cli/module_registry.go

```diff
@@ -0,0 +1,260 @@
+func CreateModuleCommand(c *cli.Context) error {
+	moduleNameArg := c.String("name")
+	orgIDArg := c.String("org_id")
+	publicNamespaceArg := c.String("public_namespace")
+
+	client, err := NewAppClient(c)
+	if err != nil {
+		return err
+	}
+	org, err := resolveOrg(client, orgIDArg, publicNamespaceArg)
+	if err != nil {
+		return err
+	}
+	if org == nil {
+		return errors.Errorf("Unable to determine org from orgID(%q) and namespace(%q)", orgIDArg, publicNamespaceArg)
+	}
+	cwd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "failed to find the current directory")
+	}
+	// Check to make sure the user doesn't accidentally overwrite a module manifest
+	manifestFilepath := filepath.Join(cwd, defaultManifestFilename)
+	if _, err := os.Stat(manifestFilepath); err == nil {
+		return errors.Errorf("Another module's %v already exists in the current directory. Delete it and try again", defaultManifestFilename)
+	}
+
+	response, err := client.CreateModule(moduleNameArg, org.GetId())
+	if err != nil {
+		return errors.Wrap(err, "failed to register the module on app.viam.com")
+	}
+
+	fmt.Fprintf(c.App.Writer, "Successfully created '%s'.\n", response.GetModuleId())
+	if response.GetUrl() != "" {
+		fmt.Fprintf(c.App.Writer, "You can view it here: %s \n", response.GetUrl())
+	}
+
+	returnedModuleID, err := parseModuleID(response.GetModuleId())
+	if err != nil {
+		return err
+	}
+	if returnedModuleID.Namespace == "" && org.PublicNamespace != "" {
+		returnedModuleID.Namespace = org.PublicNamespace
+	}
+	emptyManifest := ModuleManifest{
+		Name:       returnedModuleID.toString(),
+		Visibility: ModuleVisibilityPrivate,
+		// This is done so that the json has an empty example
+		Models: []ModuleComponent{
+			{},
+		},
+	}
+	if err := writeManifest(manifestFilepath, emptyManifest); err != nil {
+		return err
+	}
+	fmt.Fprintf(c.App.Writer, "Configuration for the module has been written to %s\n", defaultManifestFilename)
+	return nil
+}
+
+// UpdateModuleCommand runs the command to update a module.
+// This includes updating the meta.json to include the public namespace (if set on the org).
+func UpdateModuleCommand(c *cli.Context) error {
+	publicNamespaceArg := c.String("public_namespace")
+	orgIDArg := c.String("org_id")
+	manifestPathArg := c.String("module")
+
+	client, err := NewAppClient(c)
+	if err != nil {
+		return err
+	}
+	manifestPath := defaultManifestFilename
+	if manifestPathArg != "" {
+		manifestPath = manifestPathArg
+	}
+	if _, err := os.Stat(manifestPath); err != nil {
+		return errors.Wrapf(err, "Cannot find %s", manifestPath)
+	}
+
+	manifestBytes, err := os.ReadFile(manifestPath)
+	if err != nil {
+		return err
+	}
+	var manifest ModuleManifest
+	if err := json.Unmarshal(manifestBytes, &manifest); err != nil {
+		return err
+	}
+	moduleID, err := parseModuleID(manifest.Name)
+	if err != nil {
+		return err
+	}
+	if publicNamespaceArg != "" {
+		switch moduleID.Namespace {
```

Comment:
golinter made me make this a switch statement because it doesn't like `else if` statements

Reply:
Fine as is. Seen that lint issue before... usually because of the `else` I think the alternative would be to refactor in to a function that returns in each condition to simplify the if.